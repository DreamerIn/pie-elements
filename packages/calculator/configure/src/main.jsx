import React from 'react';
import ReactDOM from 'react-dom';
import Calculator from '@pie-framework/material-ui-calculator';
import { InputRadio, TwoChoice } from '@pie-lib/config-ui';

export default class Main extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      twoChoice: '',
      model: props.model
    };
  }

  render() {
    const { basic, scientific } = this.state;
    return (
      <div>
        <TwoChoice
          header="Choose Calculator Type"
          value={this.state.twoChoice}
          onChange={twoChoice => this.setState({ twoChoice })}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }} />
          {this.state.twoChoice && <Calculator mode={this.state.twoChoice} />}
      </div>
    );
  }
}