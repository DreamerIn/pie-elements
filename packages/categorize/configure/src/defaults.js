export default {
  choices: [
    {
      id: '0',
      content: 'Choice 0'
    },
  ],
  choicesPerRow: 2,
  choicesPosition: 'below',
  choicesLabel: '',
  lockChoiceOrder: true,
  removeTilesAfterPlacing: false,
  categoriesPerRow: 2,

  categories: [
    {
      id: '0',
      label: 'Category 0',
      choices: []
    },
  ],
  correctResponse: [],
};
