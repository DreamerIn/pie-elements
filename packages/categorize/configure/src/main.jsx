import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { Tabs } from '@pie-lib/config-ui';
import Design from './design';
import Scoring from './scoring';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    imageSupport: PropTypes.object
  };

  static defaultProps = {};

  changeScoring = scoring => {
    const { onChange, model } = this.props;
    model.scoring = scoring;
    onChange(model);
  };

  render() {
    const { classes, className, model, onChange } = this.props;

    if (!model.scoring) {
      model.scoring = {
        weighting: {
          enabled: false,
          rules: []
        },
        partial: {
          enabled: false,
          rules: []
        }
      };
    }

    return (
      <div className={classNames(classes.main, className)}>
        <Tabs>
          <Design
            imageSupport={this.props.imageSupport}
            title="Design"
            model={model}
            onChange={onChange}
          />
          <Scoring
            title="Scoring"
            scoring={model.scoring}
            categories={model.categories}
            correctResponse={model.correctResponse}
            onChange={this.changeScoring}
          />
        </Tabs>
      </div>
    );
  }
}
const styles = theme => ({
  main: {}
});

const StyledMain = withStyles(styles)(Main);

class Stateful extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object.isRequired,
    imageSupport: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      model: props.model
    };
  }

  componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = model => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return (
      <StyledMain
        imageSupport={this.props.imageSupport}
        model={this.state.model}
        onChange={this.onChange}
      />
    );
  }
}

export default Stateful;
