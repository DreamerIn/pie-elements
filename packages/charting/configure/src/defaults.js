/**
 * NOTE: There's no functionality described for graphTitle,
 * rationale, scoringType, studentInstructions, teacherInstructions
 * so there's no implementation (they are only added in model)
 */


const createCategory = (label, value) => ({
  label,
  value,
  initial: true,
  interactive: true,
  editable: true,
  deletable: true
});

export default {
  model: {
    addCategoryEnabled: true,
    categoryDefaultLabel: 'Category',
    chartType: 'lineCross',
    correctAnswer: {},
    data: [],
    domain: {
      label: 'Fruits',
    },
    editCategoryEnabled: true,
    graph: {
      width: 480,
      height: 480
    },
    prompt: 'Here goes item stem!',
    rationale: 'Rationale goes here!',
    range: {
      label: 'Amount',
      max: 5.5,
      min: 0,
      labelStep: 1,
    },
    scoringType: 'partial scoring',
    title: 'This is a chart!',
  },
  configuration: {
    prompt: {
      settings: true,
      label: 'Item Stem'
    },
    rationale: {
      settings: true,
      label: 'Rationale',
      enabled: false
    },
    scoringType: {
      settings: true,
      label: 'Scoring Type'
    },
    studentInstructions: {
      settings: false,
      label: 'Student Instructions',
      enabled: true
    },
    teacherInstructions: {
      settings: true,
      label: 'Teacher Instructions',
      enabled: false
    },
    title: {
      settings: false,
      label: 'Graph Title',
      enabled: true
    }
  }
};
