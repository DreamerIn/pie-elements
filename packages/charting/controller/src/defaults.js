/**
 * NOTE: There's no functionality described for arrows, padding, labels, graphTitle,
 * rationale, scoringType, studentInstructions, teacherInstructions
 * so there's no implementation (they are only added in model)
 */

export default {
  scoringType: 'auto',
  configure: {
    graphTitle: {
      settings: false,
      label: 'Graph Title',
      enabled: true
    },
    labels: {
      settings: false,
      label: 'Labels',
      enabled: true
    },
    rationale: {
      settings: false,
      label: 'Rationale',
      enabled: true,
    },
    scoringType: {
      settings: false,
      label: 'Scoring Type',
    },
    studentInstructions: {
      settings: false,
      label: 'Student Instructions',
      enabled: true,
    },
    teacherInstructions: {
      settings: false,
      label: 'Teacher Instructions',
      enabled: false,
    },
  },
};
