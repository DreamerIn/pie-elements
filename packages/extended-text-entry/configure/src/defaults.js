export default {
  model: {
    dimensions: {
      height: 100,
      width: 500
    },
    prompt: 'This is the question prompt',
    mathInput: false,
    equationEditor: 'everything'
  },
  configuration: {
    dimensions: {
      settings: true,
      label: 'Text-Entry Display Size'
    },
    equationEditor: {
      settings: false,
      label: 'Equation Editor',
      enabled: true,
    },
    feedback: {
      settings: true,
      label: 'Feedback',
      enabled: true
    },
    mathInput: {
      settings: true,
      label: 'Student response can include math notation',
      enabled: false,
    },
    multiple: {
      settings: false,
      label: 'Multiple Parts',
      enabled: false,
    },
    studentInstructions: {
      settings: false,
      label: 'Student Instructions',
      enabled: true,
    },
    teacherInstructions: {
      settings: true,
      label: 'Teacher Instructions',
      enabled: true,
    },
  }
};
