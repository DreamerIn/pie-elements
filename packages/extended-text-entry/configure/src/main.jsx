import React from 'react';
import {
  NumberTextField,
  InputCheckbox,
  FeedbackSelector,
  InputContainer
} from '@pie-lib/config-ui';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import EditableHtml from '@pie-lib/editable-html';

const defaultFeedback = {
  type: 'default',
  default: 'Your answer has been submitted'
};

export class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      setDimensions: true
    };
    this.changeWidth = this.change('width');
    this.changeHeight = this.change('height');
  }

  change = key => (event, v) => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ [key]: v });
    onChange(model);
  };

  onPromptChange = markup => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ prompt: markup });
    onChange(model);
  };

  applyUpdate(update) {
    const { model } = this.props;
    const out = Object.assign({}, model, update);
    return out;
  }

  toggleMath = event => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ showMathInput: event.target.checked });
    onChange(model);
  };

  changeFeedback = feedback => {
    const { model, onChange } = this.props;
    const update = { ...model, feedback };
    onChange(update);
  };

  render() {
    const { model, classes } = this.props;
    return (
      <div>
        <Typography className={classes.header} variant="subheading">
          Display
        </Typography>
        <NumberTextField
          label="Width (px)"
          disabled={!this.state.setDimensions}
          value={parseInt(model.width)}
          min={100}
          max={500}
          onChange={this.changeWidth}
        />
        <NumberTextField
          label="Height (px)"
          disabled={!this.state.setDimensions}
          value={parseInt(model.height)}
          min={100}
          max={500}
          onChange={this.changeHeight}
        />
        <br />
        <InputCheckbox
          label={'Student responses can include math notation'}
          onChange={this.toggleMath}
          checked={!!model.showMathInput}
        />
        <br />
        <InputContainer label="Prompt" className={classes.promptContainer}>
          <EditableHtml markup={model.prompt} onChange={this.onPromptChange} />
        </InputContainer>

        <Typography className={classes.header} variant="subheading">
          Feedback
        </Typography>

        <FeedbackSelector
          label="When submitted, show"
          feedback={model.feedback || defaultFeedback}
          onChange={this.changeFeedback}
        />
      </div>
    );
  }
}
export default withStyles(theme => ({
  header: {
    paddingBottom: theme.spacing.unit
  },
  promptContainer: {
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
    width: '100%'
  },
  promptInput: {
    width: '100%',
    marginBottom: theme.spacing.unit
  }
}))(Main);
