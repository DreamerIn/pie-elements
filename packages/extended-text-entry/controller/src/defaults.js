export default {
  prompt: '<div>This is the question prompt</div>',
};
