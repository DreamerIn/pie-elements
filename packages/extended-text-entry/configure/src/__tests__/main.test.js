import React from 'react';
import { Main } from '../main';
import { shallow } from 'enzyme';

import {
  layout,
  settings,
} from '@pie-lib/config-ui';

import defaults from '../defaults';

jest.mock('@pie-lib/config-ui', () => ({
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  settings: {
    Panel: props => <div onChange={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn()
  }
}));

describe('Render Main Component', () => {
  let wrapper, instance, onChange;
  let model = defaults;

  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(
      <Main
        classes={{}}
        model={model}
        onChange={onChange}
        handleBoxResize={() => {}}
      />
    );

    instance = wrapper.instance();
  });

  it('Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('logic', () => {
    it('changeWidth calls onChange', () => {
      instance.changeWidth({}, 10);
      expect(onChange).toBeCalledWith(expect.objectContaining({ width: 10 }));
    });

    it('changeHeight call onChange', () => {
      instance.changeHeight({}, 10);
      console.log(onChange.mock.calls[0]);
      expect(onChange).toBeCalledWith(expect.objectContaining({ height: 10 }));
    });
  });
});
