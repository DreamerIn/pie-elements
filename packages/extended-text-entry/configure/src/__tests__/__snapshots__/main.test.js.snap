// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Component Match Snapshot 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "dimensions": Object {
            "label": "Text-Entry Display Size",
            "settings": true,
          },
          "equationEditor": Object {
            "enabled": true,
            "label": "Equation Editor",
            "settings": false,
          },
          "feedback": Object {
            "enabled": true,
            "label": "Feedback",
            "settings": true,
          },
          "mathInput": Object {
            "enabled": false,
            "label": "Student response can include math notation",
            "settings": true,
          },
          "multiple": Object {
            "enabled": false,
            "label": "Multiple Parts",
            "settings": false,
          },
          "prompt": Object {
            "enabled": true,
            "label": "Prompt",
            "settings": true,
          },
          "studentInstructions": Object {
            "enabled": true,
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "enabled": true,
            "label": "Teacher Instructions",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "configure.studentInstructions.enabled": false,
            "teacherInstructions.enabled": undefined,
          },
          "Settings": Object {
            "dimensions": undefined,
            "equationEditor": false,
            "feedback.enabled": undefined,
            "mathInput": undefined,
            "multiple.enabled": false,
            "prompt.enabled": undefined,
          },
        }
      }
      model={
        Object {
          "dimensions": Object {
            "height": 100,
            "width": 500,
          },
          "equationEditor": "everything",
          "mathInput": false,
          "prompt": "This is the question prompt",
        }
      }
      onChangeConfiguration={[Function]}
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <WithStyles(Typography)
      variant="subheading"
    >
      Display
    </WithStyles(Typography)>
    <Component
      label="Teacher Instructions"
    >
      <EditableHtml
        markup=""
        nonEmpty={false}
        onChange={[Function]}
      />
    </Component>
    <br />
    <Component
      label="Prompt"
    >
      <EditableHtml
        markup="This is the question prompt"
        nonEmpty={false}
        onChange={[Function]}
      />
    </Component>
    <WithStyles(Typography)
      variant="subheading"
    >
      Feedback
    </WithStyles(Typography)>
    <Component
      feedback={
        Object {
          "default": "Your answer has been submitted",
          "type": "default",
        }
      }
      label="When submitted, show"
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;
